osu.Game/Skinning/SkinnableDrawable.cs

```diff
@@ -12,7 +14,7 @@ namespace osu.Game.Skinning
     /// <summary>
     /// A drawable which can be skinned via an <see cref="ISkinSource"/>.
     /// </summary>
-    public partial class SkinnableDrawable : SkinReloadableDrawable
+    public partial class SkinnableDrawable : SkinReloadableDrawable, ISerialisableDrawableContainer
```

Comment:
The `ISerialisableDrawableContainer` members in here should all be explicitly implemented (i.e. not be implemented by public methods), because as I take it, nothing except the skinning management hierarchy itself should be able to call any of these methods.

Reply:
Looks better, but I think you missed one in https://github.com/ppy/osu/pull/27773/commits/2893c957b36374a6c492297d152c72a981eb7122:

```diff
diff --git a/osu.Game/Skinning/SkinnableDrawable.cs b/osu.Game/Skinning/SkinnableDrawable.cs
index 7688140176..d743e2606d 100644
--- a/osu.Game/Skinning/SkinnableDrawable.cs
+++ b/osu.Game/Skinning/SkinnableDrawable.cs
@@ -80,7 +80,7 @@ protected SkinnableDrawable(ISkinComponentLookup lookup, ConfineMode confineMode
         /// </summary>
         protected virtual bool ApplySizeRestrictionsToDefault => false;
 
-        protected override void SkinChanged(ISkinSource skin) => Reload();
+        protected override void SkinChanged(ISkinSource skin) => ((ISerialisableDrawableContainer)this).Reload();
 
         protected override void Update()
         {
@@ -112,7 +112,7 @@ protected override void Update()
             }
         }
 
-        public void Reload() => reload(CurrentSkin.GetDrawableComponent(Lookup));
+        void ISerialisableDrawableContainer.Reload() => reload(CurrentSkin.GetDrawableComponent(Lookup));
 
         private void reload(Drawable? newComponent)
         {

```

Reply:
I intentionally didn't change this one because it didn't seem meaningful? But alright.